core/Date.php

```diff
@@ -141,11 +144,17 @@ public static function factory($dateString, $timezone = null)
             $date = new Date($dateString);
         }
         $timestamp = $date->getTimestamp();
-        // can't be doing web analytics before the 1st website
-        // Tue, 06 Aug 1991 00:00:00 GMT
-        if ($timestamp < 681436800) {
-            throw self::getInvalidDateFormatException($dateString);
+    
+        $dateOfFirstWebsite = new self(self::FIRST_WEBSITE_TIMESTAMP);
+        if ($date->isEarlier($dateOfFirstWebsite)) {
```

Comment:
Any reason to not use `$timestamp < 681436800` or `$timestamp < self::FIRST_WEBSITE_TIMESTAMP`? Would make it a wee bit faster as this might be executed quite a few times. Could still create the date within the `if block` in case the date is actually earlier.

Reply:
Nope, just thought it's gonna be more readable. I'll move that to the `if` block.